docs/project_proposal.md

```diff
@@ -0,0 +1,116 @@
+# Starlark Language Server Proposal
+
+Do you wish that your editor/IDE provided fancy capabilities for [Starlark] (e.g., auto complete, go
+to definition, hover information)? If so, read on and help us get to our funding goal to make this
+project a reality.
+
+| Quick Facts  |                                                                                                                                                          |
+| ------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| What         | Implement a language server for [Starlark].                                                                                                              |
+| Why          | Fancy features like go to definition, hover information and auto complete in your favorite editor/IDE. (see [Language Server Protocol] for more details) |
+| Who          | [Chuck Grindel]                                                                                                                                          |
+| When         | As soon as we meet our funding goals.                                                                                                                    |
+| Language     | Implement the server using [Go].                                                                                                                         |
+| License      | [Apache 2.0]                                                                                                                                             |
+| GitHub Owner | <https://github.com/bazel-contrib>                                                                                                                       |
+| Contribute   | Help us reach our funding goals by [contributing].                                                                                                       |
+
+## 💻 What is a language server?
+
+A language server is a command-line executable that implements the [Language Server Protocol] (LSP).
+It interacts with a client in your editor/IDE to provide the smarts that every developer loves and
+wishes that they had for [Starlark].
+
+## 📝 The Plan
+
+The plan consists of four phases:
+
+1. Initial server implementation plus support for go to definition and find references.
+1. Support hover information.
+1. Support auto completion.
+1. Support signature help.
+
+Once we meet our initial funding goals, other features that are on the roadmap are support
+for
+
+- linting
+- formatting
+- code lens (provide contextual information and actions)
+- code folding
+- rename variables/functions
+
```

Comment:
Some org such as SourceGraph could be interested in LSIF / SCIP integration.

Especially in bigger repositories where you have to index a lot of files to support "Find references", having the ability to pre-seed the LSP index with LSIF might be necessary.